### ISSUE TITLE
Would you consider adding a contrib/ folder?

### ISSUE THREAD

**Issue by Russell91** [naftaliharris/python2.8#50]
Would you consider adding a contrib/ folder with a few useful-to-haves like get-pip.py, perhaps a port of google's pytype for python2.8, or other tools? The python source is normally meant for ppl experimenting with the compiler and therefore lacks a few of the things you'd want in a download like this that is intended for a wider audience. Makes sense if you're not considering this for the time being though - obviously you've gone out of your way to not add anything outside the python3 PEPs.

If nothing else, having a anaconda style wrapper around this (even if not provided by continuum) would make it a lot easier for people to get started, and therefore for ppl to share their tools based on it.

**Comment by naftaliharris** [comment]
> If nothing else, having a anaconda style wrapper around this (even if not provided by continuum) would make it a lot easier for people to get started, and therefore for ppl to share their tools based on it.

That's a reasonable idea. It's honestly not very user-friendly at all to need to download the source, build it yourself, install it, install pip, install your dependencies... I would ideally like to figure out how to make installers or something like that. It's an open question for me.

**Comment by Russell91** [comment]
Right, it's kind of painful but rather straightforward to get your own ppa on debian. People would then be able to do sudo apt-get add-apt-reposityory ... ; sudo apt-get install python2.8. Homebrew is also pretty easy. This would probably be your best bet. I think get-pip.py would be a minimal thing that would be really useful to include for people though. Perhaps something with virtualenv as well. I do like keeping this project clean though (as ppl are angry enough as it is for stepping on their turf). Perhaps a wrapper github project that uses this one as upstream could be a good idea.

**Comment by naftaliharris** [comment]
> Between the two of those you'd have 90% of people covered though.

Makes sense.

> I think get-pip.py would be a minimal thing that would be really useful to include for people though.

Yeah, I agree. There's even some related work on this in Python 3, (https://docs.python.org/3/library/ensurepip.html), though I haven't investigated it much yet.

**Comment by Russell91** [comment]
Actually, it just occurred to me that anaconda itself is a perfect host for the compiled python2.8! Users can just run cp python2.8/python ~/anaconda/bin/python and they'll have pip, numpy, etc. all ready and available. Nothing else would really be required. To make this even easier, and more safe, one could maintain a host a separate Anaconda shell compiler, where people could just install a new version of anaconda that would clone this repository, compile it, and then put the python in the right place. This would also allow addition of pytype2.8, as I talked about above, so ppl could get started with type checking as I talked about above. Of course we could let continuum maintain the system as well if they eventually took interest. But their EULA appears to allow this:

https://docs.continuum.io/anaconda/eula

Btw, for what it's worth, I would vote for the "Nython is not official python" name. Kind of celebrates the fact that it's okay to think differently, and we're not so much opposed to the python3 people. It's just that our community seems to be really struggling without the support of a compiler team to add things like static types, and as programmers, we can't afford to wait around another 8 years for the PSF to help us. The scientific community in particular, seems to be making no progress at all towards python3 adoption. That's why I think anaconda could be a really good combination though, as many of us use it anyway.
